tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetStorage.scala

```diff
@@ -0,0 +1,100 @@
+package com.linagora.tmail.james.jmap.publicAsset
+
+import java.io.InputStream
+import java.net.URI
+import java.util.UUID
+
+import com.google.common.collect.ImmutableList
+import com.linagora.tmail.james.jmap.publicAsset.ImageContentType.ImageContentType
+import eu.timepit.refined
+import eu.timepit.refined.api.{Refined, Validate}
+import org.apache.http.client.utils.URIBuilder
+import org.apache.james.blob.api.BlobId
+import org.apache.james.jmap.api.model.IdentityId
+import org.apache.james.jmap.api.model.Size.Size
+import org.apache.james.jmap.core.AccountId
+import org.apache.james.mailbox.model.ContentType
+
+import scala.util.Try
+
+object PublicAssetIdFactory {
+  def generate(): PublicAssetId = PublicAssetId(UUID.randomUUID())
+}
+
+case class PublicAssetId(value: UUID) {
+  def asString(): String = value.toString
+}
+
+object PublicURI {
+  def fromString(value: String): Either[Throwable, PublicURI] = Try(new URI(value))
+    .map(PublicURI.apply)
+    .toEither
+
+  def from(assetId: PublicAssetId, account: AccountId, publicUriPrefix: URI): PublicURI = {
```

Comment:
Can you change `AccountId` to `Username`? cf https://github.com/linagora/tmail-backend/issues/1045#issuecomment-2123868651